### PR TITLE
kealib: migrate to Conan v2

### PR DESCRIPTION
**Pull request by valgur**
TODO

- [ ] `undefined reference to `H5::` etc
- [ ] v1: need to add `-std=c++11` to library build

### CODE REVIEW

**Comment by valgur** [comment]
Closing temporarily to avoid unnecessary load on the CI. Will reopen when I'm actively working on the PR again.

File: recipes/kealib/all/conanfile.py [hunk 1]
```diff
@@ -1,74 +1,105 @@
...
+    def _min_cppstd(self):
+        return 11
 
     @property
-    def _build_subfolder(self):
-        return "build_subfolder"
+    def _compilers_minimum_version(self):
+        return {
+            "gcc": "6",
+        }
```
Comment (uilianries) [inline recipes/kealib/all/conanfile.py:41]:
```suggestion
```
It's not required and can be bypassed by CMAKE_CXX_STANDARD instead.

File: recipes/kealib/all/conanfile.py [hunk 2]
```diff
@@ -1,74 +1,105 @@
...
 
     def validate(self):
-        if not (self.options["hdf5"].enable_cxx and self.options["hdf5"].hl):
+        if self.settings.compiler.cppstd:
+            check_min_cppstd(self, self._min_cppstd)
+        minimum_version = self._compilers_minimum_version.get(str(self.settings.compiler), False)
+        if minimum_version and Version(self.settings.compiler.version) < minimum_version:
+            raise ConanInvalidConfiguration(
+                f"{self.ref} requires C++{self._min_cppstd}, which your compiler does not support."
+            )
```
Comment (uilianries) [inline recipes/kealib/all/conanfile.py:66]:
```suggestion
```

File: recipes/kealib/all/conanfile.py [hunk 3]
```diff
@@ -1,74 +1,105 @@
...
-        self._cmake.configure(build_folder=self._build_subfolder)
-        return self._cmake
+        get(self, **self.conan_data["sources"][self.version], strip_root=True)
+
+    def generate(self):
+        tc = CMakeToolchain(self)
+        tc.cache_variables["HDF5_USE_STATIC_LIBRARIES"] = not self.dependencies["hdf5"].options.shared
+        tc.cache_variables["HDF5_PREFER_PARALLEL"] = self.dependencies["hdf5"].options.parallel
+        tc.cache_variables["HDF5_THREADSAFE"] = self.dependencies["hdf5"].options.get_safe("threadsafe", False)
+        tc.cache_variables["LIBKEA_WITH_GDAL"] = False
```
Comment (uilianries) [inline recipes/kealib/all/conanfile.py:79]:
```suggestion
        tc.cache_variables["LIBKEA_WITH_GDAL"] = False
        # INFO: kealib uses C++11 but does not configure in cmake: https://github.com/ubarsc/kealib/pull/48
        tc.variables["CMAKE_CXX_STANDARD"] = self.settings.get_safe("compiler.cppstd", self._min_cppstd)
```

**Review by uilianries — COMMENTED** [review]
Please, let's keep GCC 5 working along the package.

**Review by RubenRBS — APPROVED** [review]

**Comment by valgur** [comment]
@uilianries Do you have any idea what's going on here?
```
Installing (downloading, building) binaries...
kealib/1.4.15: Retrieving package 4dcd3ad87c9c2d2b0621973e9aa23dd3f3dcc606 from remote 'c3i_PR-18680' 
kealib/1.4.15: WARN: Package binary is corrupted, removing: 4dcd3ad87c9c2d2b0621973e9aa23dd3f3dcc606
ERROR: Binary package not found: 'kealib/1.4.15:4dcd3ad87c9c2d2b0621973e9aa23dd3f3dcc606'
```

File: recipes/kealib/all/conanfile.py [hunk 4]
```diff
@@ -1,74 +1,101 @@
...
-        self.requires("hdf5/1.12.0")
+        self.requires("hdf5/1.14.3", transitive_headers=True, transitive_libs=True)
 
     def validate(self):
-        if not (self.options["hdf5"].enable_cxx and self.options["hdf5"].hl):
+        if self.settings.compiler.cppstd:
+            check_min_cppstd(self, self._min_cppstd)
+        hdf5_opts = self.dependencies["hdf5"].options
+        if not (hdf5_opts.enable_cxx and hdf5_opts.hl):
             raise ConanInvalidConfiguration("kealib requires hdf5 with cxx and hl enabled.")
```
Comment (uilianries) [inline recipes/kealib/all/conanfile.py:58]:
```suggestion
            raise ConanInvalidConfiguration(f"{self.ref} requires dependencies options -o 'hdf5/*:enable_cxx=True' -o 'hdf5/*:hl=True'")
```

**Review by uilianries — APPROVED** [review]
LGTM

**Review by perseoGI — APPROVED** [review]
LGTM

**Pull request merged**